src/function-builder-detectors/netlify-lambda.js

```diff
@@ -1,12 +1,20 @@
 const { existsSync, readFileSync } = require('fs')
 const execa = require('execa')
 
-module.exports = function() {
-  if (!existsSync('package.json')) {
-    return false
+/**
+ * Function builder detector for netlify-lambda.
+ *
+ * @param {object} [packageSettings] npm package.json format, if not provided, defaults to reading package.json from the file system.
+ */
+module.exports = function(packageSettings) {
```

Comment:
Instead of changing the API for the detector, can we extract the relevant part to a function and test only that function?